src/components/Search/CompareWithBase.tsx

```diff
@@ -129,87 +122,85 @@ function CompareWithBase({
     }),
   };
 
-  useEffect(() => {
-    setStagingBaseRevs(baseRevs);
-    setStagingNewRevs(newRevs);
-  }, [baseRevs, newRevs]);
-
-  useEffect(() => {
-    if (newInProgress) {
-      setDisplayedRevisionsNewRevs(newInProgressRevs);
-    } else {
-      setDisplayedRevisionsNewRevs(newStagingRevs);
-    }
-
-    if (baseInProgress) {
-      setDisplayedRevisionsBaseRevs(baseInProgressRevs);
-    } else {
-      setDisplayedRevisionsBaseRevs(baseStagingRevs);
-    }
-  }, [
-    newInProgress,
-    newInProgressRevs,
-    newStagingRevs,
-    baseInProgress,
-    baseInProgressRevs,
-    baseStagingRevs,
-  ]);
-
   const toggleIsExpanded = () => {
     setExpanded(!expanded);
   };
+
   const handleCancelBase = () => {
-    setInProgressBaseRevs([]);
-    setInProgressBase(false);
-    dispatch(clearCheckedRevisionforType({ searchType: 'base' }));
+    setInProgressBaseRevs(baseStagingRevs);
   };
-
   const handleCancelNew = () => {
-    setInProgressNewRevs([]);
-    setInProgressNew(false);
-    dispatch(clearCheckedRevisionforType({ searchType: 'new' }));
+    setInProgressNewRevs(newStagingRevs);
   };
 
   const handleSaveBase = () => {
     setStagingBaseRevs(baseInProgressRevs);
-    handleCancelBase();
   };
-
   const handleSaveNew = () => {
     setStagingNewRevs(newInProgressRevs);
-    handleCancelNew();
   };
 
   const handleEditBase = () => {
     setInProgressBaseRevs(baseStagingRevs);
-    setInProgressBase(true);
   };
-
   const handleEditNew = () => {
     setInProgressNewRevs(newStagingRevs);
-    setInProgressNew(true);
   };
 
   const handleRemoveRevisionBase = (item: Changeset) => {
     const revisionsBase = [...baseInProgressRevs];
     revisionsBase.splice(baseInProgressRevs.indexOf(item), 1);
     setInProgressBaseRevs(revisionsBase);
   };
-
   const handleRemoveRevisionNew = (item: Changeset) => {
     const revisionsNew = [...newInProgressRevs];
     revisionsNew.splice(newInProgressRevs.indexOf(item), 1);
     setInProgressNewRevs(revisionsNew);
   };
 
-  const handleSearchResultsToggleBase = (toggleArray: Changeset[]) => {
-    setInProgressBaseRevs(toggleArray || []);
-    setInProgressBase(true);
+  const handleItemToggleInChangesetList = ({
```

Comment:
It will probably be worth extracting this again, because Carla will want to reuse this in CompareOverTime. I'm not sure yet if this should be a hook or just a utility function.
I can do it after #629.